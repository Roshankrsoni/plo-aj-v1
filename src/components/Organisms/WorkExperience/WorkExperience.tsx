import React, { useMemo } from "react";
import "./WorkExperience.css";
import { useGlobalContext } from "../../../context/GlobalContext";
import moment from "moment/moment";
import { ExperienceDetails } from "../../../types/profileDataTypes";

const WorkExperience = () => {
  const { profileData } = useGlobalContext();
  const { experience } = profileData;

  const calculatedExperience = useMemo((): Array<ExperienceDetails> => {
    return experience.map((exp): ExperienceDetails => {
      const fromDate = moment(exp.fromDate);
      const toDate = exp.toDate !== "Present" ? moment(exp.toDate) : moment();

      const years = toDate.diff(fromDate, "year");
      fromDate.add(years, "years");
      const months = toDate.diff(fromDate, "months");
      fromDate.add(months, "months");

      let yearsOfExp = "";
      let monthsOfExp = "";

      if (years > 1) {
        yearsOfExp = `${years} years`;
      } else if (years == 1) {
        yearsOfExp = `${years} year`;
      }

      if (months > 1) {
        monthsOfExp = `${months} months`;
      } else if (months == 1) {
        monthsOfExp = `${years} month`;
      }

      exp.totalYears = `${yearsOfExp} ${monthsOfExp}`;
      return exp;
    });
  }, [experience]);

  function getFromToDates(fromDate: string, toDate: string): string {
    return `${fromDate.substring(3)} - ${
      toDate.includes("Present") ? toDate : toDate.substring(3)
    }`;
  }

  return (
    <div className="col-md-7 page-1 text-left section work-experience ">
      <div className="section-title h2 bold">Work Experience</div>

      <div className="subsection">
        {calculatedExperience.map((experienceDetail, key) => {
          return (
            <div key={key} className="subsection-data">
              <span className="h3 grey1 bold">{experienceDetail.name}</span>
              <div className="label grey3 light mt-1 letterspacing-2 row no-gutters align-items-center">
                <span className="pr-3">
                  {getFromToDates(
                    experienceDetail.fromDate,
                    experienceDetail.toDate
                  )}
                </span>
                <span className="dot"></span>
                <span className="pl-3">{experienceDetail.totalYears}</span>
              </div>
              <div className="mt-3 grey-1 body-text">
                {experienceDetail.summary}
              </div>
            </div>
          );
        })}
      </div>
    </div>
  );
};

export default WorkExperience;
