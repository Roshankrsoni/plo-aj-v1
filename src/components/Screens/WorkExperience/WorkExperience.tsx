import React, { useMemo } from "react";
import { useGlobalContext } from "../../../context/GlobalContext";
import moment from "moment/moment";
import { ExperienceDetails } from "../../../types/profileDataTypes";
import SectionVisibilityHOC from "../../Organisms/SectionVisibilityHOC/SectionVisibilityHOC";
import T from "../../../translations/en_IN";
import constants from "../../../helpers/constants";
const WorkExperience = () => {
  const { profileData } = useGlobalContext();
  const { experience } = profileData;

  const calculatedExperience = useMemo((): Array<ExperienceDetails> => {
    return experience.map((exp): ExperienceDetails => {
      const fromDate = moment(exp.fromDate);
      const toDate = exp.toDate !== "Present" ? moment(exp.toDate) : moment();

      const years = toDate.diff(fromDate, "year");
      fromDate.add(years, "years");
      const months = toDate.diff(fromDate, "months");
      fromDate.add(months, "months");

      let yearsOfExp = "";
      let monthsOfExp = "";

      if (years > 1) {
        yearsOfExp = `${years} years`;
      } else if (years == 1) {
        yearsOfExp = `${years} year`;
      }

      if (months > 1) {
        monthsOfExp = `${months} months`;
      } else if (months == 1) {
        monthsOfExp = `${years} month`;
      }

      exp.totalYears = `${yearsOfExp} ${monthsOfExp}`;
      return exp;
    });
  }, [experience]);

  function getFromToDates(fromDate: string, toDate: string): string {
    return `${fromDate.substring(3)} - ${
      toDate.includes("Present") ? toDate : toDate.substring(3)
    }`;
  }

  return (
    <>
      <div className="section-title h2 bold">{T.WORK_EXPERIENCE}</div>
      <div className="subsection">
        {calculatedExperience.map((experienceDetail, key) => {
          return (
            <div key={key} className="subsection-data">
              <span className="h3 grey1 bold">{experienceDetail.name}</span>
              <div className="label grey5 light mt-1 letterspacing-2 row no-gutters align-items-center">
                <span className="pr-3">
                  {getFromToDates(
                    experienceDetail.fromDate,
                    experienceDetail.toDate
                  )}
                </span>
                <span className="dot"></span>
                <span className="pl-3">{experienceDetail.totalYears}</span>
              </div>
              <div className="mt-3 grey-1 body-text">
                {experienceDetail.summary}
              </div>
            </div>
          );
        })}
      </div>
    </>
  );
};

export default SectionVisibilityHOC(
  WorkExperience,
  constants.classNames.WORK_EXPERIENCE
);
