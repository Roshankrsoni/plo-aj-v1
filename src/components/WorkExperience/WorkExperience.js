import "./WorkExperience.css";

function WorkExperience(props) {
  return (
    <div
      className={
        "show-on-scroll col-md-7 page-1 text-left section work-experience " +
        props.class
      }
    >
      <div className="section-title grey4 h2 bold">Work Experience</div>

      <div className="subsection">
        {props.dbData.experience.map((ele, key) => {
          return <div key={key} className="subsection-data">
            <span className="h3 grey1 bold">{ele.name}</span>
            <div className="label grey3 light mt-1 letterspacing-2 row no-gutters align-items-center">
              <span className="pr-3">{ele.fromToDates}</span>
              <span className="dot"></span>
              <span className="pl-3">{ele.totalYears}</span>
            </div>
            <div className="mt-3 grey-1 body-text">{ele.summary}</div>
            {/*  <a className="know-more-button mt-4"><span className="red pr-2">Know more</span> <img src={KnowMoreButton} alt="know more buton" /></a> */}
          </div>
        })}
      </div>
    </div>
  );
}

export default WorkExperience;
