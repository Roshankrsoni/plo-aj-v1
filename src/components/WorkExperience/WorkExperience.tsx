import React from "react";
import "./WorkExperience.css";
import { useGlobalContext } from "../../context/GlobalContext";

const WorkExperience = () => {
  const { profileData } = useGlobalContext();
  const { experience } = profileData;

  return (
    <div className="col-md-7 page-1 text-left section work-experience ">
      <div className="section-title h2 bold">Work Experience</div>

      <div className="subsection">
        {experience.map((experienceDetail, key) => {
          return (
            <div key={key} className="subsection-data">
              <span className="h3 grey1 bold">{experienceDetail.name}</span>
              <div className="label grey3 light mt-1 letterspacing-2 row no-gutters align-items-center">
                <span className="pr-3">{experienceDetail.fromToDates}</span>
                <span className="dot"></span>
                <span className="pl-3">{experienceDetail.totalYears}</span>
              </div>
              <div className="mt-3 grey-1 body-text">
                {experienceDetail.summary}
              </div>
            </div>
          );
        })}
      </div>
    </div>
  );
};

export default WorkExperience;
