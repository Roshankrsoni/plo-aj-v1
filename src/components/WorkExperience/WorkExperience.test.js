import React from 'react';
import { shallow } from 'enzyme';
import WorkExperience from './WorkExperience';

import dummyData from '../TestData/testData.json'

describe('MyComponent', () => {
    it('should render correctly in "debug" mode', () => {
        const component = shallow(<WorkExperience dbData={dummyData} />);
        expect(component).toMatchSnapshot();
    });
});