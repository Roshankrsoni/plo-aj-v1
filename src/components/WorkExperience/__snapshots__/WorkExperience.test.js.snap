// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly in "debug" mode 1`] = `
<div
  className="show-on-scroll col-md-7 page-1 text-left section work-experience undefined"
>
  <div
    className="section-title grey4 h2 bold"
  >
    Work Experience
  </div>
  <div
    className="subsection"
  >
    <div
      className="subsection-data"
      key="0"
    >
      <span
        className="h3 grey1 bold"
      >
        Thoughtworks
      </span>
      <div
        className="label grey3 light mt-1 letterspacing-2 row no-gutters align-items-center"
      >
        <span
          className="pr-3"
        >
          Currently Working
        </span>
        <span
          className="dot"
        />
        <span
          className="pl-3"
        >
          2 Months
        </span>
      </div>
      <div
        className="mt-3 grey-1 body-text"
      >
        Thoughtworks has been my dream company since college days because of its work culture. It is widely know for the intellectual minds who are solving complex problem using technology. I am really excited to work on new challenges here in Thoughtworks.
      </div>
    </div>
    <div
      className="subsection-data"
      key="1"
    >
      <span
        className="h3 grey1 bold"
      >
        Deloitte
      </span>
      <div
        className="label grey3 light mt-1 letterspacing-2 row no-gutters align-items-center"
      >
        <span
          className="pr-3"
        >
          2016-2021
        </span>
        <span
          className="dot"
        />
        <span
          className="pl-3"
        >
          4 Years
        </span>
      </div>
      <div
        className="mt-3 grey-1 body-text"
      >
        In Deloitte, I started with a public sector project for State of Michigan where we built a custom application on salesforce using salesforce lightning components. Along with the project I worked on several Firm initiatives built on React Js - One of the major project being Elixir where we build a React app for non developers similar to wordpress/squarespace having the drag and drop functionality to generate salesforce components. Due to the open culture in Deloitte, I also got the opportunity to work on design projects which involved in-house web/mobile app design, several posters, mailers, infographics, brochure designs and branding. 
      </div>
    </div>
    <div
      className="subsection-data"
      key="2"
    >
      <span
        className="h3 grey1 bold"
      >
        Cognizant
      </span>
      <div
        className="label grey3 light mt-1 letterspacing-2 row no-gutters align-items-center"
      >
        <span
          className="pr-3"
        >
          2014-2016
        </span>
        <span
          className="dot"
        />
        <span
          className="pl-3"
        >
          2 Years
        </span>
      </div>
      <div
        className="mt-3 grey-1 body-text"
      >
        Initially started my career in Cognizant as a SAP BASIS consultant. but due to my interest in web technologies I switched to a UI Developer role and worked on various projects using tools and technologies like Angular JS 1.x., React Js, Jquery, Bootstrap, Git etc.
      </div>
    </div>
  </div>
</div>
`;
